deps/rabbitmq_federation/src/rabbit_federation_exchange_link.erl

```diff
@@ -504,15 +504,16 @@ consume_from_upstream_queue(
     #upstream{prefetch_count = Prefetch,
               expires        = Expiry,
               message_ttl    = TTL,
-              ha_policy      = HA} = Upstream,
+              queue_type     = QueueType} = Upstream,
     #upstream_params{x_or_q = X,
                      params = Params} = UParams,
     Q = upstream_queue_name(name(X), vhost(Params), DownXName),
     Args = [A || {_K, _T, V} = A
                      <- [{<<"x-expires">>,          long,    Expiry},
                          {<<"x-message-ttl">>,      long,    TTL},
-                         {<<"x-ha-policy">>,        longstr, HA},
-                         {<<"x-internal-purpose">>, longstr, <<"federation">>}],
+                         {<<"x-internal-purpose">>, longstr, <<"federation">>},
+                         {<<"x-queue-type">>,       longstr, atom_to_binary(QueueType)}
```

Comment:
If `queue-type` is not set, this will set it to `<<"undefined">>` or something like that. Should we be more defensive here?

Reply:
The default (`classic`) is set here: https://github.com/rabbitmq/rabbitmq-server/blob/exchange-federation-qq/deps/rabbitmq_federation/src/rabbit_federation_upstream.erl#L139.